rootfs/scheduler/__init__.py

```diff
@@ -639,29 +639,40 @@ def _set_image_secret(self, data, namespace, **kwargs):
         Take registry information and set as an imagePullSecret for an RC / Deployment
         http://kubernetes.io/docs/user-guide/images/#specifying-imagepullsecrets-on-a-pod
         """
+        secret_name = settings.REGISTRY_SECRET_PREFIX
         registry = kwargs.get('registry', {})
-        if not registry:
+        if not registry and settings.REGISTRY_LOCATION == 'on-cluster':
+            return
+        elif registry:
+            # try to get the hostname information
+            hostname = registry.get('hostname', None)
```

Comment:
I had kept it thinking it might be useful in future when we want to expand to supporting per app private registry basis